src/tests/asyncTests.ts

```diff
@@ -215,12 +208,12 @@ export class AsyncTests {
     const { apexTestClassIdSet, testResults, globalTests } =
       await this.buildAsyncTestResults(apexTestResults);
 
-    let outcome = testRunSummary.Status;
+    let outcome = testRunSummary?.Status ?? 'Unknown';
```

Comment:
with this change, the TestResult summary field will be 'Unknown' if the state is not one of Completed states? In that case what is the point of creating the TestResult?

Reply:
I believe there is some unknown edge case where this function is being called with a undefined test summary, but has reportable results. By assigning defaults to test summary fields when test summary is undefined instead of throwing allows the user to at least see some results.

Another option would be to allow all output to be created, with the defaults and then throw.

Reply:
> this function is being called with a undefined test summary, but has reportable results
This seems to be a bug then?

Reply:
The undefined is one of the states that `checkRunStatus` can return. The anecdotal use case is that the tests are still in progress, but I have yet to reproduce that case. The streaming client [subscribe](https://github.com/forcedotcom/salesforcedx-apex/blob/f78cfc7627f7d1dc7fde73b94d9911410f16dcfc/src/streaming/streamingClient.ts#L168) function is the one that waits for test to complete and checkRunStatus reports on current state.

I found it more reasonable to allow a undefined test summary to be used in reporting results rather than throw an error and lose any aggregated results. One of these commands can run for hours, so a failure ensure the user repeating the command.

Reply:
what I am confused about though is what a summary result with Unknown and with results mean to the user?

Reply:
Have we seen cases other than the one mentioned in the issue (where the tests were still in "Processing"), where we have received Undefined and there were results?